hadoop-ozone/recon/src/main/resources/webapps/recon/ozone-recon-web/src/v2/components/multiSelect/columnTag.tsx

```diff
@@ -0,0 +1,60 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from "react";
+import { Tag } from "antd";
+import { createPortal } from "react-dom";
+
+export type TagProps = {
+  label: string;
+  closable: boolean;
+  tagRef: React.RefObject<HTMLDivElement>;
+  onClose: (arg0: string) => void;
+}
+
+const ColumnTag: React.FC<TagProps> = ({
+  label = '',
+  closable = true,
+  tagRef = null,
+  onClose = () => {}
+}) => {
```

Comment:
Default props should be defined outside the functional component. This ensures that the default values are consistent and not recreated on every render.

Reply:
This is actually not required, because as of React 18 onwards React is going to deprecate default props.
This is the proposal from React team for the deprecation: https://github.com/reactjs/rfcs/blob/createlement-rfc/text/0000-create-element-changes.md
Currently it will give warning, but eventually we need to shift to this way for compatibility.
Also React has internal optimizations so it doesn't affect performance compared to declaring the default values separately.
Here is the React docs on a similar performance related question:
https://legacy.reactjs.org/docs/hooks-faq.html#are-hooks-slow-because-of-creating-functions-in-render